### PR TITLE
Sprint 38 TLT-1155 Post user and role to api, handle results

### PR DESCRIPTION
**Pull request by rascalking**
Still needs to be updated to Vittorio's new warning UI, but it's functional.

### CODE REVIEW

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 1]
```diff
@@ -14,26 +14,64 @@
...
         };
-        $scope.closeSuccess = function(index) {
-            $scope.successes.splice(index, 1);
+        $scope.addUserToCourse = function(searchTerm, user){
+            var url = djangoUrl.reverse('icommons_rest_api_proxy',
+                                        ['api/course/v2/course_instances/'
+                                         + $scope.courseInstanceId + '/people/']);
+            $http.post(url, user)
+                .success(function(data, status, headers, config, statusText) {
+                    if (data.detail) {
```
Comment (eparker71) [inline course_info/static/course_info/js/controllers/PeopleController.js]: does data.detail indicate a partial error? I'm just checking, I haven't see this before.

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 2]
```diff
@@ -14,26 +14,64 @@
...
+                        // TODO - put more details into this warning, like
+                        //        and user details.
```
Comment (eparker71) [inline course_info/static/course_info/js/controllers/PeopleController.js]: is the idea here to look at data.details and see if there are any inconsistencies in the data? then add those to partialFailures?

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 3]
```diff
@@ -187,7 +247,7 @@
                               'icommons_rest_api_proxy',
                               ['api/course/v2/course_instances/' + id + '/']);
                 $http.get(url)
-                    .success(function(data, status, headers, config) {
+                    .success(function(data, status, headers, config, statusText) {
```
Comment (eparker71) [inline course_info/static/course_info/js/controllers/PeopleController.js]: looks like statusText was added but not used?

File: course_info/static/course_info/js/controllers/PeopleController.js [hunk 4]
```diff
@@ -172,11 +233,10 @@
...
             return /^.*feed$/.test(data) ? 'Registrar Added' : 'Manually Added';
         },
-        $scope.selectRole = function(roleId, roleName){
-            $scope.selectedrole = roleId;
-            $('#select-role-btn-id').text(roleName);
+        $scope.selectRole = function (role) {
+            $scope.selectedRole = role;
         };
-        $scope.setTitle = function(id) {
+        $scope.setTitle = function (id) {
```
Comment (eparker71) [inline course_info/static/course_info/js/controllers/PeopleController.js]: extra space after fucntion

**Comment by eparker71** [comment]
This looks good :+1: I think the one thing to fix is the extra space that might blow up. The others comments are just clarifications.

File: course_info/static/course_info/js/controllers/PeopleController.js (hunk 2 above)
Comment (rascalking) [inline course_info/static/course_info/js/controllers/PeopleController.js]: Yup.  Currently, we get a note in there that the API was unable to add to Canvas.  Need to add logic in there to only add the partialFailure if we know the course has a Canvas site.  But that's the general idea.

File: course_info/static/course_info/js/controllers/PeopleController.js (hunk 3 above)
Comment (rascalking) [inline course_info/static/course_info/js/controllers/PeopleController.js]: Yeah, I'd rather have access to all the params being passed to my function.

File: course_info/static/course_info/js/controllers/PeopleController.js (hunk 4 above)
Comment (rascalking) [inline course_info/static/course_info/js/controllers/PeopleController.js]: On purpose.

Comment (rascalking) [inline course_info/static/course_info/js/controllers/PeopleController.js]: Meh, on further googling, I'll revert this.  Looks like my personal preference for space-if-arguments, no-space-if-no-arguments is just a personal preference.

**Pull request merged**